drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -129,14 +129,70 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF700X_ON_QSPI
 )
 
 # RPU FW patch binaries based on the selected configuration
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SYSTEM_MODE
-  -DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin
-)
-
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_RADIO_TEST
-  -DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin
-)
+if(CONFIG_NRF700X_SYSTEM_MODE)
+  set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin)
+elseif (CONFIG_NRF700X_RADIO_TEST)
+  set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin)
+elseif (CONFIG_NRF700X_SCAN_ONLY)
+  set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin)
+else()
+  # Error
+  message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SCAN_ONLY
-  -DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin
-)
+zephyr_compile_definitions(
+  -DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
+)
+
+if(CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
+  message(STATUS "nRF WiFi FW patch binary will be stored in external flash")
+
+  dt_chosen(ext_flash_dev PROPERTY nordic,pm-ext-flash)
+  if(DEFINED ext_flash_dev)
+    get_filename_component(qspi_node ${ext_flash_dev} DIRECTORY)
+  else()
+    dt_nodelabel(qspi_node NODELABEL "qspi")
+  endif()
+  if(DEFINED qspi_node)
+    dt_reg_addr(xip_addr PATH ${qspi_node} NAME qspi_mm)
+    if(NOT DEFINED xip_addr)
+      message(FATAL_ERROR "Unable to find XIP area in dts")
+    endif()
+  else()
+    message(FATAL_ERROR "QSPI node not found in dts")
+  endif()
+
+  dt_nodelabel(nrf70_fw_node NODELABEL "nrf70_fw_partition")
+  if(NOT DEFINED nrf70_fw_node)
+    message(FATAL_ERROR "nrf70_fw_partition node not found in dts")
+  endif()
+  dt_prop(nrf70_fw_offset PATH ${nrf70_fw_node} PROPERTY reg INDEX 0)
+  if(NOT DEFINED nrf70_fw_offset)
+    message(FATAL_ERROR "nrf70_fw_partition.reg not found in dts")
+  endif()
+  math(EXPR NRF70_PATCH_ADDR "${xip_addr} + ${nrf70_fw_offset}")
+
+  find_package(Python3)
```

Comment:
needs to use ncs/zephyr python not find it again

Reply:
So, directly use `${Python3_EXECUTABLE}` ?